packages/aws-cdk/lib/api/bootstrap/bootstrap-template.json

```diff
@@ -0,0 +1,394 @@
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    },
+    "HasCustomFileAssetsBucketName": {
+      "Fn::Not": [
+        {
+          "Fn::Equals": [
+            "",
+            {
+              "Ref": "FileAssetsBucketName"
+            }
+          ]
+        }
+      ]
+    },
+    "CustomKmsKeyIdNotPassed": {
+      "Fn::Equals": [
+        "",
+        {
+          "Ref": "CustomKmsKeyId"
+        }
+      ]
+    },
+    "HasCustomContainerAssetsRepositoryName": {
+      "Fn::Not": [
+        {
+          "Fn::Equals": [
+            "",
+            {
+              "Ref": "CustomContainerAssetsRepositoryName"
+            }
+          ]
+        }
+      ]
+    }
+  },
+  "Resources": {
+    "FileAssetsBucketEncryptionKey": {
+      "Type": "AWS::KMS::Key",
+      "Properties": {
+        "KeyPolicy": {
+          "Statement": [
+            {
+              "Action": [
+                "kms:Create*", "kms:Describe*", "kms:Enable*", "kms:List*", "kms:Put*",
+                "kms:Update*", "kms:Revoke*", "kms:Disable*", "kms:Get*", "kms:Delete*",
+                "kms:ScheduleKeyDeletion", "kms:CancelKeyDeletion", "kms:GenerateDataKey"
+              ],
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Fn::Sub": "arn:${AWS::Partition}:iam::${AWS::AccountId}:root"
+                }
+              },
+              "Resource": "*"
+            },
+            {
+              "Action": [
+                "kms:Decrypt",    "kms:DescribeKey",     "kms:Encrypt",
+                "kms:ReEncrypt*", "kms:GenerateDataKey*"
+              ],
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Fn::Sub": "arn:${AWS::Partition}:iam::${AWS::AccountId}:root"
+                }
+              },
+              "Resource": "*",
+              "Condition": {
+                "StringEquals": {
+                  "kms:ViaService": [
+                    { "Fn::Sub": "s3.${AWS::Region}.amazonaws.com" }
+                  ]
+                }
+              }
+            },
+            {
+              "Action": [
+                "kms:Decrypt",    "kms:DescribeKey",     "kms:Encrypt",
+                "kms:ReEncrypt*", "kms:GenerateDataKey*"
+              ],
+              "Effect": "Allow",
+              "Principal": {
+                "AWS": {
+                  "Fn::Sub": "${PublishingRole.Arn}"
+                }
+              },
+              "Resource": "*"
+            }
+          ]
+        }
+      },
+      "Condition": "CustomKmsKeyIdNotPassed"
+    },
+    "StagingBucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": {
+          "Fn::If": [
+            "HasCustomFileAssetsBucketName",
+            { "Fn::Sub": "${FileAssetsBucketName}" },
+            { "Fn::Sub": "cdk-bootstrap-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}" }
```

Comment:
hnb659fds?

Reply:
This is just a semi-random string to reduce the chance of name conflicts.